src/dishka/integrations/litestar.py

```diff
@@ -94,3 +99,46 @@ def setup_dishka(container: AsyncContainer, app: Litestar) -> None:
         app.asgi_handler,
     )
     app.state.dishka_container = container
+
+
+class DishkaMiddleware(ASGIMiddleware):
+    scopes = (ScopeType.HTTP, ScopeType.WEBSOCKET)
+    async def handle(
+            self, scope: Scope, receive: Receive,
+            send: Send, next_app: ASGIApp,
+    ) -> None:
+        if scope.get("type") not in (ScopeType.HTTP, ScopeType.WEBSOCKET):
+            await next_app(scope, receive, send)
+            return
+
+        if scope.get("type") == ScopeType.HTTP:
+            request: Request = Request(scope)
+            context = {Request: request}
+            di_scope = DIScope.REQUEST
+            async with request.app.state.dishka_container(
+                    context,
+                    scope=di_scope,
+            ) as request_container:
+                request.state.dishka_container = request_container
+                await next_app(scope, receive, send)
+
+        elif scope.get("type") == ScopeType.WEBSOCKET:
+            websocket: WebSocket = WebSocket(scope)
+            context = {WebSocket: websocket}
+            di_scope = DIScope.SESSION
+            async with websocket.app.state.dishka_container(
+                    context,
+                    scope=di_scope,
+            ) as request_container:
+                websocket.state.dishka_container = request_container
+                await next_app(scope, receive, send)
+
+
+class DishkaPlugin(InitPlugin):
```

Comment:
can we add plugin automatically inside `setup_dishka`?

Reply:
I dont think so, 

the current api is
```
    app = litestar.Litestar()
    container = make_async_container(provider)
    setup_dishka(container, app)
```

and this PR just adds the possibility to do:
```
    container = make_async_container(provider)
    app = litestar.Litestar(plugins=[DishkaPlugin(container=container)])
```

I dont know a litestar's built-in way to add a plugin other than using the plugins kwarg unfortunately.